llvm/lib/Analysis/Loads.cpp

```diff
@@ -319,11 +319,14 @@ bool llvm::isDereferenceableAndAlignedInLoop(
   const SCEV *MaxBECount =
       Predicates ? SE.getPredicatedConstantMaxBackedgeTakenCount(L, *Predicates)
                  : SE.getConstantMaxBackedgeTakenCount(L);
+  const SCEV *SymbolicMaxBECount =
+      Predicates ? SE.getPredicatedConstantMaxBackedgeTakenCount(L, *Predicates)
```

Comment:
s/Constant/Symbolic/? Perhaps pass ExitKind to getPredicatedBackedgeTakenCount?

Reply:
But this isn't the symbolic maximum, right? The documentation says:

```
    /// A constant which provides an upper bound on the exact trip count.
    ConstantMaximum,
    /// An expression which provides an upper bound on the exact trip count.
    SymbolicMaximum,
```

Surely, it should be called ConstMaxBECount?